### PR TITLE
MODLD-601: LCCN deduplication configuration setting

### PR DESCRIPTION
**Pull request by AndreiBordak**

### CODE REVIEW

File: src/main/resources/application.yml [hunk 1]
```diff
@@ -82,6 +82,7 @@ mod-linked-data:
   cache:
     ttl:
       spec-rules: ${CACHE_TTL_SPEC_RULES:18000000}
+      settings-entries: ${CACHE_TTL_SETTINGS_ENTRIES:18000000}
```
Comment (AndreiBordak) [inline src/main/resources/application.yml:85]: spring.cache.type=none can be used to disable all caches in karate tests
@pkjacob

Comment (askhat-abishev) [inline src/main/resources/application.yml:85]: ```CACHE_TTL_SETTINGS_ENTRIES``` should be added to README and ModuleDescriptor files

File: src/main/java/org/folio/linked/data/validation/dto/LccnUniquenessValidator.java [hunk 2]
```diff
@@ -41,6 +54,14 @@ public boolean isValid(ResourceRequestDto resourceRequestDto, ConstraintValidato
     return true;
   }
 
+  private boolean isValidationDisabled() {
+    try {
+      return !settingsService.isSettingEnabled(DUPLICATE_CHECK_SCOPE, DUPLICATE_CHECK_KEY, DUPLICATE_CHECK_PROPERTY);
+    } catch (Exception e) {
+      return true;
```
Comment (askhat-abishev) [inline src/main/java/org/folio/linked/data/validation/dto/LccnUniquenessValidator.java:61]: maybe add some log here?

**Review by askhat-abishev — APPROVED** [review]

File: src/main/resources/application.yml [hunk 3]
```diff
@@ -81,8 +81,8 @@ mod-linked-data:
       instance-ingress: ${KAFKA_INVENTORY_INSTANCE_INGRESS_EVENT_TOPIC:inventory.instance_ingress}
   cache:
     ttl:
-      spec-rules: ${CACHE_TTL_SPEC_RULES:18000000}
-      settings-entries: ${CACHE_TTL_SETTINGS_ENTRIES:18000000}
+      spec-rules: ${CACHE_TTL_SPEC_RULES_MS:18000000}
```
Comment (askhat-abishev) [inline src/main/resources/application.yml:84]: :+1:

File: src/main/java/org/folio/linked/data/service/SettingsService.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package org.folio.linked.data.service;
+
+import java.util.Optional;
+import org.folio.linked.data.domain.dto.SettingsItem;
+
+public interface SettingsService {
+
+  Optional<SettingsItem> getSetting(String scope, String key);
```
Comment (pkjacob) [inline src/main/java/org/folio/linked/data/service/SettingsService.java:8]: Any need to expose this method in the interface? I believe only `isSettingEnabled` need to be exposed in the interface and `getSetting` can be a private method in the impl class.

Comment (AndreiBordak) [reply src/main/java/org/folio/linked/data/service/SettingsService.java:8]: just wanted to make it public for future usages, but yes, we ain't gonna need it now

**Review by pkjacob — APPROVED** [review]

File: src/main/java/org/folio/linked/data/service/SettingsService.java (hunk 4 above)
Comment (AndreiBordak) [reply src/main/java/org/folio/linked/data/service/SettingsService.java]: changed to private

File: src/main/java/org/folio/linked/data/validation/dto/LccnUniquenessValidator.java [hunk 5]
```diff
@@ -41,6 +54,15 @@ public boolean isValid(ResourceRequestDto resourceRequestDto, ConstraintValidato
     return true;
   }
 
+  private boolean isValidationDisabled() {
+    try {
+      return !settingsService.isSettingEnabled(DUPLICATE_CHECK_SCOPE, DUPLICATE_CHECK_KEY, DUPLICATE_CHECK_PROPERTY);
+    } catch (Exception e) {
+      log.warn("An exception occurred during settings service call", e);
```
Comment (AndreiBordak) [inline src/main/java/org/folio/linked/data/validation/dto/LccnUniquenessValidator.java]: decided to change it to 'error'

**Pull request merged**